internals/secrethub/tpl/template.go

```diff
@@ -0,0 +1,106 @@
+package tpl
+
+import (
+	"fmt"
+
+	"github.com/secrethub/secrethub-cli/internals/tpl"
+)
+
+// Template is a template containing secret paths.
+//
+// Secret paths can be supplied between brackets:
+// {{ path/to/secret }}.
+// Secret paths can contain variables. These can be
+// supplied using: ${ var }.
+// Combined this can look like:
+// {{ ${app}/db/secret }}
+type Template struct {
```

Comment:
I found myself being pretty confused about the name of this. Especially because this `Template` contains a map of other `Template` structs. Is there a way in which we can distinguish between the two? Or is there a specific reason for this naming?

Reply:
`internals/tpl` is very generic; It allows parsing and substitution of any value between any delimiter. 
`internals/secrethub/tpl` is more specific; It allows parsing of secret paths given by `{{` `}}` delimiters, which can contain variables given by `${` `}` delimiters.

`internals/secrethub/tpl` is not specifically for run templates, it can be used to inject files as well.

So you could say the `internals/secrethub/tpl` is specifically a SecretHub template, while `internals/tpl` can be any template.

---

Having this extra information, what would be a name that explains this best? Should we put more emphasis on the `secrethub` import path of `internals/secrethub/tpl`?